### PR TITLE
Remove deprecated fields

### PR DESCRIPTION
**Pull request by benashz**
In preparation for the major version bump, all deprecated fields related to < vault-1.2 have been removed.

As this PR is quite large, all related updates to the upgrade guide will be done in a separate PR.

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-vault/blob/master/CHANGELOG.md):

```release-note
TBD
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'
=== RUN   TestDataSourceIdentityEntityName
--- PASS: TestDataSourceIdentityEntityName (2.18s)
=== RUN   TestDataSourceIdentityEntityAlias
--- PASS: TestDataSourceIdentityEntityAlias (2.04s)
=== RUN   TestDataSourceIdentityGroupName
--- PASS: TestDataSourceIdentityGroupName (2.05s)
=== RUN   TestDataSourceIdentityGroupAlias
--- PASS: TestDataSourceIdentityGroupAlias (2.10s)
[...]
--- PASS: TestResourceToken_basic (2.45s)
=== RUN   TestResourceToken_import
--- PASS: TestResourceToken_import (2.78s)
=== RUN   TestResourceToken_full
--- PASS: TestResourceToken_full (2.63s)
=== RUN   TestResourceToken_lookup
--- PASS: TestResourceToken_lookup (2.65s)
=== RUN   TestResourceToken_expire
--- PASS: TestResourceToken_expire (16.30s)
=== RUN   TestResourceToken_renew
--- PASS: TestResourceToken_renew (27.59s)
=== RUN   TestAccTransitCacheConfig
--- PASS: TestAccTransitCacheConfig (7.09s)
=== RUN   TestTransitSecretBackendKey_basic
--- PASS: TestTransitSecretBackendKey_basic (3.86s)
=== RUN   TestTransitSecretBackendKey_rsa4096
--- PASS: TestTransitSecretBackendKey_rsa4096 (6.67s)
=== RUN   TestTransitSecretBackendKey_import
--- PASS: TestTransitSecretBackendKey_import (2.85s)
=== RUN   TestExpandAuthMethodTune
--- PASS: TestExpandAuthMethodTune (0.00s)
=== RUN   TestFlattenAuthMethodTune
--- PASS: TestFlattenAuthMethodTune (0.00s)
=== RUN   TestValidateNoTrailingSlash
--- PASS: TestValidateNoTrailingSlash (0.00s)
PASS
ok  	github.com/hashicorp/terraform-provider-vault/vault	710.069s

CircleCI received exit code 0
...
```

### CODE REVIEW

File: vault/resource_gcp_auth_backend_role.go [hunk 1]
```diff
@@ -36,11 +36,6 @@ func gcpAuthBackendRoleResource() *schema.Resource {
 			Optional: true,
 			ForceNew: true,
 		},
-		"project_id": {
```
Comment (tomhjp) [inline vault/resource_gcp_auth_backend_role.go:39]: Why is this disappearing?

File: vault/resource_generic_secret.go [hunk 2]
```diff
@@ -47,12 +47,6 @@ func genericSecretResource() *schema.Resource {
 				Sensitive:    true,
 			},
 
-			"allow_read": {
```
Comment (tomhjp) [inline vault/resource_generic_secret.go:50]: Not sure why this is deleted?

**Review by tomhjp — COMMENTED** [review]
Code changes look good to me, although I think I'm missing some context on why a couple of the fields are being deleted.

One question on the upgrade experience: if users update deprecated fields to use the new field name, will the terraform plan report any diff?

File: vault/resource_generic_secret.go (hunk 2 above)
Comment (benashz) [reply vault/resource_generic_secret.go:50]: It was previously marked as `Removed`, but was not  properly removed in https://github.com/hashicorp/terraform-provider-vault/pull/1175/files#diff-bcbec1f006c1e5fef492a7d3fb86cdc7c589b4cc0166230a8ed27b933030fdddL54

File: vault/resource_gcp_auth_backend_role.go (hunk 1 above)
Comment (benashz) [reply vault/resource_gcp_auth_backend_role.go:39]: It was previously marked as Removed, but was not properly removed in:

https://github.com/hashicorp/terraform-provider-vault/pull/1175/files#diff-733d5d028c1c84512a7399c45cae4cb5a1c7cdddaf281c4a3def2100b7761111L43

**Review by tomhjp — APPROVED** [review]
Discussed the upgrade path offline and not much we can do to avoid diff noise. LGTM!

**Pull request merged**